config/commands.js

```diff
@@ -676,18 +676,21 @@ var commands = exports.commands = {
 					break;
 
 				case 'moves':
+					var move = {};
+					for (var i in searches[search]) {
+						move[i] = Tools.getMove(i);
+						if (!move[i].exists) return this.sendReplyBox("'" + move[i] + "' is not a known move.");
+					}
 					for (var mon in dex) {
 						var template = Tools.getTemplate(dex[mon].id);
 						if (!template.learnset) template = Tools.getTemplate(template.baseSpecies);
 						if (!template.learnset) continue;
 						for (var i in searches[search]) {
-							var move = Tools.getMove(i);
```

Comment:
Having to copy the whole move object every time is what was slowing it down so much. Sense the only property that is being used in this command is ```move.id``` I suppose I could create something like ```moveId = move[i].id;``` . I'm not sure that's any more clear then the way I have it now though.